### DISCUSSION TITLE
Unable to get row information as text in Shortcuts

### DISCUSSION THREAD

**Discussion by phkus** [vincode-io/Zavala#261] [Support]
I am unable to work with the "row" variables in Shortcuts: They work fine when used directly in other actions in the same shortcut, but I cannot get information about the row as text in the "Show result", "Stop and output", or "Set variable" actions. S. the first screenshot to show that these actions don't show any output (in this case the EntityID), even though the information is in fact in the variable, as the second screenshot shows.

The reason I need this information as text output is that the shortcut is part of an Applescript, where I would reuse the EntityID later. Specifically, I'm trying to create a script that exports all notes from the Mac app Tinderbox to Zavala for mobile access. But for that I need to return the EntityID of the last created row, so that I can pass it as input to the next iteration of the shortcut (to specify where the next row should be added).

![image](https://github.com/user-attachments/assets/53d7915c-114b-459d-a712-33d26224e719)
![image](https://github.com/user-attachments/assets/248eeb4f-f76c-40ce-a1bb-5b6ca64f1d49)

**Comment by vincode-io** [comment]
I think what you are trying to do is turn an Entity ID into a String that you could then use later as an Entity ID again. I don't think this would be possible as a String isn't an Entity ID and I don't know of anyway to convert it using Shortcuts.

You can assign an Entity ID to a variable. It just isn't going to be the display representation (String).

<img width="768" alt="Screenshot 2024-11-24 at 1 09 21 PM" src="https://github.com/user-attachments/assets/be14ca83-de2d-4d26-9337-a404b4a23342">

Would it be possible to script OPML exports from Tinderbox and import the OPML into Zavala? It seems like a more straightforward approach to me, unless there is something more you are trying to do.
